src/NorthwindWebsite/Core/EmailSender/EmailSender.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.AspNetCore.Identity.UI.Services;
+using Microsoft.Extensions.Options;
+using SendGrid;
+using SendGrid.Helpers.Mail;
+
+namespace NorthwindWebsite.Core.EmailSender;
+
+public class EmailSender : IEmailSender
+{
+    public EmailSender(IOptions<AuthMessageSenderOptions> optionsAccessor)
+    {
+        Options = optionsAccessor.Value;
+    }
+
+    public AuthMessageSenderOptions Options { get; } //set only via Secret Manager
+
+    public Task SendEmailAsync(string email, string subject, string message)
+    {
+        return Execute(Options.SendGridApiKey, subject, message, email);
+    }
+
+    public Task Execute(string apiKey, string subject, string message, string email)
+    {
+        var client = new SendGridClient(apiKey);
+        var msg = new SendGridMessage()
+        {
+            From = new EmailAddress("evg.tsarkov@gmail.com", "Northwind Training App"),
```

Comment:
Never hardcode any emails inside of your code, move to app settings, please (probably, email subject from settings is also a good idea)